### PR TITLE
Add parameter set and variable names to output metadata

### LINKED ISSUE #24: Add parameter set to output metadata

### LINKED ISSUE #25: Add parameter for output precision

### PR DESCRIPTION
**Pull request by arbennett**
- [x] closes #24 
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master | flake8 --diff``

### CODE REVIEW

File: metsim/metsim.py [hunk 1]
```diff
@@ -66,12 +67,20 @@
...
+                  'standard_name': 'air_temperature'},
+         'vapor_pressure': {'units': 'mbar', 'long_name': 'vapor pressure',
+                            'standard_name': 'vapor_pressure'},
+         'tskc': {'units': 'fraction', 'long_name': 'cloud fraction',
+                  'standard_name': 'cloud_fraction'},
+         'rel_humid': {'units': '%', 'long_name': 'relative humidity',
+                       'standard_name': 'relative_humidity'},
          '_global': {'conventions': '1.6', 'title': 'Output from MetSim',
                      'institution': 'University of Washington',
                      'source': 'metsim.py',
```
Comment (jhamman) [inline metsim/metsim.py]: I think it probably makes sense to make all of these OrderedDict type for consistency.

File: metsim/metsim.py [hunk 2]
```diff
@@ -275,6 +281,16 @@ def setup_output(self, prototype: xr.Dataset=None):
         dims = ('time', ) + self.domain['mask'].dims
         coords = {'time': times, **self.domain.mask.coords}
         self.output = xr.Dataset(coords=coords)
+        for k, v in self.params.items():
+            # Need to convert some parameters to strings
+            if k in ['start', 'stop', 'annual', 'mtclim_swe_corr']:
+                v = str(v)
+            # Don't include complex types
+            if type(v) is list:
```
Comment (jhamman) [inline metsim/metsim.py]: 1) I think it is better to use `isinstance` when checking types.
2) We should check generally for `v` being an iterable. For example, a `tuple` should be treated in the same way here.

File: metsim/metsim.py [hunk 3]
```diff
@@ -275,6 +281,16 @@ def setup_output(self, prototype: xr.Dataset=None):
...
+                v = ', '.join(v)
+            elif type(v) in [pd.DatetimeIndex, OrderedDict]:
+                continue
```
Comment (jhamman) [inline metsim/metsim.py]: so, we're just skipping these?

File: metsim/metsim.py (hunk 1 above)
Comment (arbennett) [inline metsim/metsim.py]: I'm not sure I understand why this would change anything in the output?

File: metsim/metsim.py (hunk 2 above)
Comment (arbennett) [inline metsim/metsim.py]: I take back my thumbs up.  Strings are iterable...

Comment (jhamman) [inline metsim/metsim.py]:
```Python
if not isinstance(v, str) and hasattr(v, '__iter__'):
    # or similar
```

?

File: metsim/metsim.py (hunk 1 above)
Comment (jhamman) [inline metsim/metsim.py]: Just so we get consistent ordering in the output file.

Comment (arbennett) [inline metsim/metsim.py]: Fixed in lines below.

**Pull request merged**